src/vivarium_public_health/population/data_transformations.py

```diff
@@ -165,6 +163,21 @@ def rescale_binned_proportions(
 
         pop_data.loc[max_bin.index, "age_end"] = age_end
         pop_data.loc[padding_bin.index, "age_start"] = age_end
+        # Update age to be within bounds
+        pop_data.loc[max_bin.index, "age"] = float(
+            (
+                pop_data.loc[max_bin.index, "age_start"].iloc[0]
+                + pop_data.loc[max_bin.index, "age_end"].iloc[0]
+            )
+            / 2
+        )
+        pop_data.loc[padding_bin.index, "age"] = float(
+            (
+                pop_data.loc[padding_bin.index, "age_start"].iloc[0]
+                + pop_data.loc[padding_bin.index, "age_end"].iloc[0]
+            )
+            / 2
+        )
 
     return pop_data[col_order].sort_values(_SORT_ORDER).reset_index(drop=True)
```

Comment:
I know this isn't specific to this PR, but it looks like whatever is going on in this method has deviated from the docstring. i.e. the docstring says it returns "Table with the same columns as `pop_data` where all bins outside the range (age_start, age_end) have been discarded", but it looks like it still returns rows outside the range as well.

I feel like there's a bug floating around in here somewhere...